safe/gui/tools/multi_exposure_dialog.py

```diff
@@ -87,6 +108,219 @@ def __init__(self, parent=None):
+            index = self.list_layers_in_map_report.indexFromItem(items[0])
+            index = index.row()
+            if index == 0:
+                self.move_up.setEnabled(False)
+                self.move_down.setEnabled(True)
+            elif index == self.list_layers_in_map_report.count() - 1:
+                self.move_up.setEnabled(True)
+                self.move_down.setEnabled(False)
+            else:
+                self.move_up.setEnabled(True)
+                self.move_down.setEnabled(True)
+        else:
+            self.move_up.setEnabled(False)
+            self.move_down.setEnabled(False)
+
+    def _tree_selection_changed(self):
+        """Selection has changed in the tree."""
+        self.add_layer.setEnabled(len(self.tree.selectedItems()) >= 1)
+
+    def _populate_reporting_tab(self):
+        """Populate trees about layers."""
+        self.tree.clear()
+        self.add_layer.setEnabled(False)
+        self.remove_layer.setEnabled(False)
+        self.move_up.setEnabled(False)
+        self.move_down.setEnabled(False)
+        self.tree.setColumnCount(1)
+        self.tree.setRootIsDecorated(False)
+        self.tree.setHeaderHidden(True)
+
+        analysis_branch = QTreeWidgetItem(
+            self.tree.invisibleRootItem(), [FROM_ANALYSIS['name']])
+        analysis_branch.setFont(0, bold_font)
+        analysis_branch.setExpanded(True)
+        analysis_branch.setFlags(Qt.ItemIsEnabled)
+
+        if self._multi_exposure_if:
+            expected = self._multi_exposure_if.output_layers_expected()
+            for group, layers in expected.iteritems():
+                group_branch = QTreeWidgetItem(analysis_branch, [group])
+                group_branch.setFont(0, bold_font)
+                group_branch.setExpanded(True)
+                group_branch.setFlags(Qt.ItemIsEnabled)
+
+                for layer in layers:
+                    layer = definition(layer)
+                    if layer.get('allowed_geometries', None):
+                        item = QTreeWidgetItem(
+                            group_branch, [layer.get('name')])
+                        item.setData(
+                            0, LAYER_ORIGIN_ROLE, FROM_ANALYSIS['key'])
+                        item.setData(0, LAYER_PARENT_ANALYSIS_ROLE, group)
+                        item.setData(
+                            0, LAYER_PURPOSE_KEY_OR_ID_ROLE, layer['key'])
+                        item.setFlags(Qt.ItemIsEnabled | Qt.ItemIsSelectable)
+
+        canvas_branch = QTreeWidgetItem(
+            self.tree.invisibleRootItem(), [FROM_CANVAS['name']])
+        canvas_branch.setFont(0, bold_font)
+        canvas_branch.setExpanded(True)
+        canvas_branch.setFlags(Qt.ItemIsEnabled)
+
+        # List layers from the canvas
+        loaded_layers = QgsMapLayerRegistry.instance().mapLayers().values()
+        canvas_layers = iface.mapCanvas().layers()
+        flag = setting('visibleLayersOnlyFlag', True, bool)
```

Comment:
You don't need to set the default value, since it's already set in the `default_settings.py`. Anyway, not mandatory, but changing in one place is better.

Reply:
Does it means it's checking that the default value is set automatically if it doesn't exist in the QSettings?

Should we add a lookup in default_settings.py when we use the `settings` function too?

Reply:
> Does it means it's checking that the default value is set automatically if it doesn't exist in the QSettings?

Yes

> Should we add a lookup in default_settings.py when we use the settings function too?

Ideally, yes. I have a wish list to make each setting has more verbose description/definition. So that we can understand, what is what in the setting.